src/main/py/frontend/frontend.py

```diff
@@ -0,0 +1,218 @@
+        width_entry.pack()
+
+        # Create the height textbox
+        height_label = tk.Label(window, text="Height:")
+        height_label.pack()
+        height_entry = tk.Entry(window)
+        height_entry.pack()
+
+        # Create the wait time textbox
+        wait_label = tk.Label(window, text="Wait Time:")
+        wait_label.pack()
+        wait_entry = tk.Entry(window)
+        wait_entry.pack()
+
+        button = tk.Button(
+            window,
+            text="Commission Art Piece",
+            command=lambda: self.commission_art_piece(
+                width_entry.get(), height_entry.get(), wait_entry.get(), window
+            ),
+        )
+        button.pack()
+
+        # Create the commission requests
+        commission_requests_label = tk.Label(window, text="Commission Requests:")
+        commission_requests_label.pack()
+        self.update_commissions(window)
+        self.peer.gui_callback = lambda: self.update_commissions(window)
+        self.update_commissions(window)
+
+    def commission_art_piece(self, width_entry, height_entry, wait_entry, window):
+        """Call commission art piece and display the result in the GUI"""
+        width = float(width_entry)
+        height = float(height_entry)
+        wait_time = float(wait_entry)
+        commission = asyncio.create_task(
+            self.peer.commission_art_piece(width, height, wait_time)
+        )
+        # Clear the window
+        for widget in window.winfo_children():
+            widget.destroy()
+        # Show a loading animation
+        loading_label = tk.Label(window, text="Loading...")
+        loading_label.pack()
+
+        async def wait_and_complete(window):
+            nonlocal commission
+            commission = await commission
+            await asyncio.sleep(wait_time)
+            loading_label.destroy()
+            complete_label = tk.Label(window, text="Commission Complete")
+            complete_label.pack()
+            image = self.peer.inventory.commission_canvases[commission.key]
+            # Resize the image if needed
+            image = image.resize((300, 300))
+            # Convert the PIL image to a Tkinter-compatible format
+            image_tk = ImageTk.PhotoImage(image)
+            # Create a Label widget to display the image
+            image_label = tk.Label(window, image=image_tk)
+            image_label.image = image_tk
+            image_label.pack()
+            # Add button to reset the gui labelled "Back to commission page"
+            reset_button = tk.Button(
+                window, text="Reset GUI", command=lambda: self.reset_gui(window)
+            )
+            reset_button.pack()
+
+        asyncio.create_task(wait_and_complete(window))
+
+    def create_gui(self):
+        """Create the GUI for the peer"""
+        # def run_tkinter_mainloop():
+        window = tk.Tk()
+        window.geometry("1000x500")  # Set the window size to 500x500
+
+        # Create the width textbox
```

Comment:
There's a lot of duplicated code between here and `reset_gui()`. Seems like a helper function could be utilized to remove that duplication.